### PR TITLE
privacy by default for room dir

### LINKED ISSUE #6348: consider defaulting the public room dir to be private
is require CS auth and not exposed over SS. Easier to go from private to public than vice versa; should default to assuming private use cases.

### PR DESCRIPTION
**Pull request by neilisfragile**
Ensure that the the default settings for the room directory are that the it is hidden from public view by default. More context [here](https://matrix.org/blog/2019/11/09/avoiding-unwelcome-visitors-on-private-matrix-servers).
fixes #6348

### CODE REVIEW

**Comment by richvdh** [comment]
fixes #6348? please try to remember to link to the issue, it saves a bunch of searching.

**Comment by anoadragon453** [comment]
This is waiting on a test to prove that the directory can actually be made private.

**Comment by neilisfragile** [comment]
New test to prove ```allow_public_rooms_over_federation```
Test for ```allow_public_rooms_without_auth``` already exists [here](https://github.com/matrix-org/synapse/blob/e7777f3668d09c87335830f785f42c851827b497/tests/rest/client/v1/test_rooms.py#L1018)

File: tests/federation/transport/test_server.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Matrix.org Foundation
```
Comment (anoadragon453) [inline tests/federation/transport/test_server.py]:
```suggestion
# Copyright 2019 The Matrix.org Foundation C.I.C.
```

File: UPGRADE.rst [hunk 2]
```diff
@@ -75,6 +75,23 @@ for example:
...
+In an attempt to configure Synapse in a privacy preserving way, the default
+behaviours of ``allow_public_rooms_without_auth`` and
+``allow_public_rooms_over_federation`` have been inverted. This means that by
+default, only authenticated users querying the Client/Server API will be able
+to query the room directory, and relatedly that the server will not share
+room directory information with other servers over federation.
+
+If your installation does not explicitly set these settings one way or the other
+and you want either setting to be ``true`` then it will necessary to update
+``homeserver.yaml`` accordingly.
```
Comment (anoadragon453) [inline UPGRADE.rst]:
```suggestion
your homeserver configuration file accordingly.
```

**Comment by neilisfragile** [comment]
Ensure that the the default settings for the room directory are that  it is hidden from public view by default.

**Pull request merged**